test/Transforms/memory_banking_multi_config.mlir

```diff
@@ -0,0 +1,168 @@
+// RUN: circt-opt %s -memory-banking="factors=2 dimensions=1" --canonicalize --split-input-file | FileCheck %s
```

Comment:
What does `factors=2 dimensions=1` here really mean? Is there documentation for these attributes?

Reply:
`factors=2 dimensions=1` means we want to bank the memory references by a factor of 2 at dimension 1, right? Or am I misunderstanding the point you are confused about?

Reply:
Ah, so this applies to every memory that does not have the factors already included.

Can a user also pass multiple factors/dimensions, e.g., `-memory-banking="factors=[2,4] dimensions=[0,1]"

Reply:
> so this applies to every memory that does not have the factors already included.

yes!

> Can a user also pass multiple factors/dimensions, e.g., `-memory-banking="factors=[2,4] dimensions=[0,1]"

yep! and this is what this PR is meant for